stout/flags.cc

```diff
@@ -24,29 +23,80 @@ void Parser::AddAllOrExit(google::protobuf::Message* message) {
   for (int i = 0; i < descriptor->field_count(); i++) {
     const auto* field = descriptor->field(i);
 
-    const auto& flag = field->options().GetExtension(stout::v1::flag);
+    // We need this descriptor for the subcommand's logic.
+    const google::protobuf::OneofDescriptor* real_oneof_field =
+        field->real_containing_oneof();
+
+    // Check if a message given by someone contains oneof field which
+    // has the `subcommand` name.
+    if (real_oneof_field != nullptr
+        && real_oneof_field->name() == "subcommand") {
+      // Run through all fields of oneof and make sure that all of them
+      // have `stout.v1.subcommand` extension.
+      for (int i = 0; i < real_oneof_field->field_count(); ++i) {
```

Comment:
This will be redundant because you'll already check each field as part of the `for` loop above.

Instead let's check if were in a `oneof` and if we are, check if it's named `subcommand` and if its not, add an error. And of course also check for the `stout:v1::subcommand` stuff like you're already doing.